Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -684,6 +687,35 @@ extension SwiftLanguageServer {
     }
   }
 
+  public func documentFormatting(_ req: Request<DocumentFormattingRequest>) {
+    guard let file = req.params.textDocument.uri.fileURL else {
+      req.reply(nil)
+      return
+    }
+    let options = req.params.options
+    self.queue.async {
+      var config = SwiftFormatConfiguration.Configuration()
```

Comment:
When invoked from the LSP, should the formatting action respect the `.swift-format.json` config file that applies to the file in the request? (Right now, we look in the same directory as the source file and then up through parent directories.)

I'm not sure what the expected behavior is for LSP formatting if you have an external config file and the options passed in the request conflict with those.

But, the `SwiftFormatter` API doesn't do anything to load that configuration; it's the `swift-format` command line tool that does it. So if the LSP needs to also look up the configuration, rather than duplicate that file lookup here, we should expose a helper API in `SwiftFormatConfiguration` that loads the configuration given the path to a particular source file so that other clients can reuse it easily (and so that if we change that logic later, all clients get matching behavior automatically).

Reply:
I would expect that we would respect the .swift-format.json configuration in general, but we should have control of this in case we need to mediate some kind of editor-level settings.  I think exposing a helper on `SwiftFormatConfiguration` would be great.

Reply:
Maybe we should use the config file if it exists, and fall back on values provided by the lsp if it doesn't exist?

Reply:
I changed it so that it tries to load config from file, and if that fails uses values provided by the lsp.
It relies on changes from the https://github.com/apple/swift-format/pull/140 pull request